rclrs/src/wait/guard_condition.rs

```diff
@@ -0,0 +1,149 @@
+use crate::rcl_bindings::*;
+use crate::{Context, RclrsError, ToResult};
+
+use std::sync::{atomic::AtomicBool, Arc, Mutex, MutexGuard};
+
+/// A struct for encapsulating a guard condition - a waitable trigger
```

Comment:
Personally, I was a bit confused when I first encountered guard conditions. A description like this may help others:

```suggestion
/// A waitable entity used for waking up a wait set manually.
///
/// If a wait set that is currently waiting on events should be interrupted from a separate thread, this can be done
/// by adding an `Arc<GuardCondition>` to the wait set, and calling `trigger()` on the same `GuardCondition` while
/// the wait set is waiting.
///
/// The guard condition may be reused multiple times, but like other waitable entities, can not be used in
/// multiple wait sets concurrently.
```

Reply:
I think there might have been a git error – it's not in the latest revision yet.